src/DelegateDecompiler/Processor.cs

```diff
@@ -1095,8 +1098,13 @@ static Expression[] GetArguments(ProcessorState state, MethodBase m)
             return mArgs;
         }
 
-        static Expression BuildMethodCallExpression(MethodInfo m, Address instance, Expression[] arguments)
+        private static Expression BuildMethodCallExpression(MethodInfo m, Address instance, Expression[] arguments)
         {
+            var concreteInstance = instance.DiscardConversion();
+            if (DelegateDecompiler.DecompileExtensions.ConcreteCalls.ContainsKey(new Tuple<object, MethodInfo>(concreteInstance, m)))

```

Comment:
Here we also check if a concrete call on the virtual chain has been performed for this particular instance